### ISSUE TITLE
[DevTools Bug]: Hook parsing failed for default Next.js configuration

### ISSUE THREAD

**Issue by rossng** [facebook/react#25427]
### Website or app

https://github.com/vercel/next.js/tree/canary/examples/blog-starter

### Repro steps

1. Create an instance of the official Next.js blog starter template, e.g. `yarn create next-app --example blog-starter blog-starter`
2. Run `yarn dev` (or equivalent) inside the created project.
3. Go to the webpage (normally `http://localhost:3000`) and open the React devtools
4. Enable 'Always parse hook names from source' in the devtool options
5. Observe the error 'Hook parsing failed':
    <img width="669" alt="image" src="https://user-images.githubusercontent.com/565371/194022450-1b32305a-b2bc-42bc-ac4d-ed389e9fdc74.png">

### How often does this bug happen?

Every time

**Comment by RZXQ** [comment]
same issue

**Comment by Astlaan** [comment]
same issue

**Comment by shynsky** [comment]
same story

**Comment by alextaymx** [comment]
I get the same issue on NExt js 12 and dev tools

**Comment by jacknight** [comment]
Does there exist any workaround for this?

**Comment by AngelMTR** [comment]
same issue nextjs 13

**Comment by sasa97972** [comment]
Same issue

**Comment by kjmph** [comment]
I got pretty frustrated not being able to debug Next.js. I didn't resolve it entirely, sorry all. However, I did notice that I can install a protocol handler for `webpack-internal` and could get further along in parsing Hooks. For instance, adding this to react-devtools causes errors that were captured in a screenshot.

```diff
diff --git a/packages/react-devtools/app.js b/packages/react-devtools/app.js
index 4b4f22ab3..31a2c7f9f 100644
--- a/packages/react-devtools/app.js
+++ b/packages/react-devtools/app.js
@@ -5,8 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const {app, BrowserWindow} = require('electron'); // Module to create native browser window.
+const {app, BrowserWindow, protocol} = require('electron'); // Module to create native browser window.
 const {join} = require('path');
+const url = require('url');
 const os = require('os');
 
 const argv = require('minimist')(process.argv.slice(2));
@@ -14,6 +15,8 @@ const projectRoots = argv._;
 
 let mainWindow = null;
 
+protocol.registerSchemesAsPrivileged([ { scheme: 'webpack-internal', privileges: { secure: true, standard: true, supportFetchAPI: true } } ]);
+
 app.on('window-all-closed', function () {
   app.quit();
 });
@@ -57,4 +60,9 @@ app.on('ready', function () {
   mainWindow.on('closed', function () {
     mainWindow = null;
   });
+
+  protocol.registerFileProtocol('webpack-internal', (request, callback) => {
+    const filePath = url.fileURLToPath('file:///Users/kjmph/test_repo/' + request.url.slice('webpack-internal://'.length));
+    callback(filePath);
+  });
 });
```

<img width="865" alt="Screenshot 2023-03-14 at 9 40 33 PM" src="https://user-images.githubusercontent.com/1075649/225184548-92cb4c49-f7c5-41da-b844-5c24ea7ca791.png">

In short, perhaps if `webpack-internal` was recognized as a protocol and react-devtools searched the CWD for files, maybe we would get a little further along. Note, Next.js has a similar issue vercel/next.js#32432 around this.

I do notice that Next.js has multiple sourceURLs and sourceMappingUrls in its compiled output, so perhaps that is why there are errors? I haven't looked into this in sometime, so could use a second set of eyes.

**Comment by kjmph** [comment]
Oh hey, look at that, it was just because I had Sentry loaded and it causes duplicate source maps? Anyway, I removed Sentry (in fact, went back to a new Next.js starter project) and the error went away (it sees a sample custom hook name):

<img width="162" alt="Screenshot 2023-03-15 at 11 03 42 PM" src="https://user-images.githubusercontent.com/1075649/225501438-e3a556cd-90c8-4634-85f6-3efac69c0f41.png">

I'll raise this in the sister issue in vercel/next.js#32432

**Comment by kjmph** [comment]
I think after quite a bit of head scratching, I put all my ruminating in that next.js issue, that I came to the conclusion that React Devtools could close this issue, as if I create a `blog-starter` I can still see the Hook names even though parsing fails:

<img width="655" alt="Screenshot 2023-03-15 at 11 37 39 PM" src="https://user-images.githubusercontent.com/1075649/225507953-566ac3b1-424c-4d2f-aab6-cf33d986e791.png">

Ultimately, I wanted to see the state variable name in React Devtools, which I thought the error message was pointing to. However, perhaps the real fix is to not throw errors when source url can not be found yet the Hook name can still be parsed from the source map? (as I believe Next.js/webpack puts the source in the JSON encoding, which is how Devtools can read this?)
